server/service/media/endpoints.js

```diff
@@ -0,0 +1,35 @@
+const { asyncMiddleware } = require('../../util/api');
+const multer = require('multer');
+const { uploadToS3 } = require('./s3');
+
+async function uploadAudio(req, res) {
+    const storage = multer.memoryStorage();
+    const upload = multer({ storage: storage });
+
+    upload.single('recording')(req, res, async err => {
+        if (err) {
+            return res
+                .status(400)
+                .json({ message: 'Upload Request Validation Failed' });
+        } else if (!req.body.name) {
+            return res
+                .status(400)
+                .json({ message: 'No recording name in request body' });
+        }
+
+        const buffer = req.file.buffer;
+        const { runId, responseId } = req.body;
+        const userId = req.session.user.id;
+
+        const s3Location = await uploadToS3(
+            `audio/${runId}/${userId}/${responseId}.mp3`,
```

Comment:
I confirmed with @rwaldron that we want to save all responses and not allow new responses to overwrite new ones. I think we'll probably need to work on some unique mp3 names here in this case? Rick is also working on some unique ID things currently, he could be good to pair with on that.

Reply:
Yes, I think you're right. @rwaldron what kind of work are you doing to make unique ID strings?